ppsspp_config.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#define PPSSPP_ARCH(PPSSPP_FEATURE) (PPSSPP_ARCH_##PPSSPP_FEATURE)
+#define PPSSPP_PLATFORM(PPSSPP_FEATURE) (PPSSPP_PLATFORM_##PPSSPP_FEATURE)
+
+// ARCH defines
+#if defined(_M_IX86) || defined(__i386__)
+    #define PPSSPP_ARCH_X86 1
+    #define PPSSPP_ARCH_32BIT 1
+    //TODO: Remove this compat define
+    #ifndef _M_IX86
+        #define _M_IX86 600
+    #endif
+#endif
+
+#if defined(_M_X64) || defined(__amd64__) || defined(__x86_64__)
+    #define PPSSPP_ARCH_AMD64 1
+    #define PPSSPP_ARCH_64BIT 1
+    //TODO: Remove this compat define
+    #ifndef _M_X64
+        #define _M_X64 1
+    #endif
+#endif
+
+#if defined(__arm__)
+    #define PPSSPP_ARCH_ARM 1
+    #define PPSSPP_ARCH_32BIT 1
+
+    #if defined(__ARM_ARCH_7__) || \
+      defined(__ARM_ARCH_7A__) || \
+      defined(__ARM_ARCH_7S__)
+        #define PPSSPP_ARCH_ARMV7 1
+    #endif
+
+    #if defined(__ARM_ARCH_7S__)
+        #define PPSSPP_ARCH_ARMV7S 1
+    #endif
+
+    #if defined(__ARM_NEON)
+        #define PPSSPP_ARCH_ARM_NEON 1
+    #endif
+
+    //TODO: Remove this compat define
+    #ifndef ARM
+        #define ARM 1
+    #endif
+#endif
+
+#if defined(__aarch64__)
+    #define PPSSPP_ARCH_ARM64 1
+    #define PPSSPP_ARCH_64BIT 1
+    #define PPSSPP_ARCH_ARM_NEON 1
+    //TODO: Remove this compat define
+    #ifndef ARM64
+        #define ARM64 1
+    #endif
+#endif
+
+#if defined(__mips__)
```

Comment:
From some quick checking, it seems mips64 may define both `__mips64__` and `__mips__`, but I'm not sure.  I was looking at some commits on llvm that seemed to indicate this:

http://llvm.org/viewvc/llvm-project/cfe/trunk/test/Preprocessor/init.c?r1=162840&r2=162858&pathrev=162858&diff_format=h

-[Unknown]